torch/_dynamo/source.py

```diff
@@ -452,6 +452,15 @@ def guard_source(self):
         return GuardSource.GLOBAL
 
 
+@dataclasses.dataclass(frozen=True)
+class DefaultDeviceSource(Source):
+    def name(self):
+        return ""
```

Comment:
Just don't define it

Reply:
Actually, we can't notimplemented this, because of provided_guarded_object.

Okay.